src/filter.jl

```diff
@@ -150,79 +140,58 @@ function signatures_from_hessian(hes::AbstractArray{T,4}) where T
                     hes[i,j,k,3]  hes[i,j,k,5]  hes[i,j,k,6]
                 ]
 
-                eigs[:] .= sort(real(eigvals(hes_slice)), rev=false)
-
-                cats[i,j,k, 1] = (abs(eigs[3] / eigs[1]) *
-                    abs(eigs[3])) * (θ(-eigs[1]) *
-                    θ(-eigs[2]) * θ(-eigs[3]))
-                cats[i,j,k, 2] = (abs(eigs[2] / eigs[1]) *
-                    xθ(1-abs(eigs[3] / eigs[1]))) *
-                    (abs(eigs[2]) * θ(-eigs[1]) * θ(-eigs[2]) )
-                cats[i,j,k, 3] = (xθ(1-abs(eigs[2] / eigs[1])) *
-                    xθ(1-abs(eigs[3] / eigs[1]))) *
-                    (abs(eigs[1]) * θ(-eigs[1]))
-                
-                sigs[i,j,k,1] = cats[i,j,k, 1] * abs(eigs[3])*θ(-eigs[1])*θ(-eigs[2])*θ(-eigs[3])
```

Comment:
I've removed some redundancy (my fault! I labeled the result of the old as script J, when it was actually script S.